cvat/apps/events/handlers.py

```diff
@@ -440,13 +441,9 @@ def filter_shape_data(shape):
             )
 
     tracks_by_type = {shape_type[0]: [] for shape_type in ShapeType.choices()}
-    for track in _annotations.get("tracks", []):
-        track_shapes = track.pop("shapes")
-        track = filter_shape_data(track)
-        track["shapes"] = []
-        for track_shape in track_shapes:
-            track["shapes"].append(filter_shape_data(track_shape))
-        tracks_by_type[track_shapes[0]["type"]].append(track)
+    for track in annotations.get("tracks", []):
+        filtered_track = filter_track(track)
+        tracks_by_type[track["shapes"][0]["type"]].append(filtered_track)
```

Comment:
Not related to this PR, but I think there may be server error when track["shapes"] is an empty array 
In theory that is possible.